command/pipelines/pipeline_execute.go

```diff
@@ -61,6 +61,28 @@ func (c *PipelineExecuteCommand) executePipeline() (gate.HttpEntity, *http.Respo
 	return entity, resp, err
 }
 
+func (c *PipelineExecuteCommand) queryExecution() ([]interface{}, *http.Response, error) {
+	config, resp, err := c.ApiMeta.GateClient.ApplicationControllerApi.GetPipelineConfigUsingGET(
+		c.ApiMeta.Context,
+		c.application,
+		c.name)
+	if err != nil {
+		return nil, nil, err
+	}
+	if resp != nil && resp.StatusCode != http.StatusOK {
+		c.ApiMeta.Ui.Error(fmt.Sprintf("Encountered an error executing pipeline, status code: %d\n", resp.StatusCode))
+		return nil, resp, err
+	}
+
+	return c.ApiMeta.GateClient.ExecutionsControllerApi.GetLatestExecutionsByConfigIdsUsingGET(
+		c.ApiMeta.Context,
+		config["id"].(string),
+		map[string]interface{}{
+			"limit":    int32(1),
```

Comment:
What if parallel executions are enabled and we get the wrong one? Should we use @joonlim's new endpoint?

Reply:
Yep SGTM worth a refactor.